qdrant-landing/content/articles/sparse-vectors.md

```diff
@@ -0,0 +1,391 @@
+            vector={
+                "text": models.SparseVector(
+                    indices=indices.tolist(),
+                    values=values.tolist()
+                )
+            }
+        )
+    ]
+)
+```
+By upserting points with sparse vectors, we prepare our dataset for rapid first-stage retrieval, laying the groundwork for subsequent detailed analysis using dense vectors. Notice that we use "text" to denote the name of the sparse vector.
+
+Those familiar with the Qdrant API will notice that the extra care taken to be consistent with the existing named vectors API -- this is to make it easier to use sparse vectors in existing codebases. As always, you're able to **apply payload filters**, shard keys, and other advanced features you've come to expect from Qdrant. To make things easier for you, the indices and values don't have to be sorted before upsert. Qdrant will sort them when the index is persisted e.g. on disk.
+
+### 4. Querying with Sparse Vectors
+
+We use the same process to prepare a query vector as well. This involves computing the vector from a query text and extracting its indices and values. We then use these details to construct a query against our collection. 
+
+```python
+# Preparing a query vector
+
+query_text = "Who was Arthur Ashe?"
+query_vec, query_tokens = compute_vector(query_text)
+query_vec.shape
+
+query_indices = query_vec.nonzero().numpy().flatten()
+query_values = query_vec.detach().numpy()[indices]
+```
+
+In this example, we use the same model for both document and query. This is not a requirement, but it's a simpler approach.
+
+### 5. Retrieving and Interpreting Results
+
+After setting up the collection and inserting sparse vectors, the next critical step is retrieving and interpreting the results. This process involves executing a search query and then analyzing the returned results.
+
+```python
+# Searching for similar documents
+result = client.search(
+        collection_name=COLLECTION_NAME,
+        query_vector=models.NamedSparseVector(
+            name="text",
+            vector=models.SparseVector(
+                indices=query_indices, 
+                values=query_values,
+            ),
+        ),
+        with_vectors=["text"],
+    )
+
+result
+```
+
+In the above code, we execute a search against our collection using the prepared sparse vector query. The `client.search` method takes the collection name and the query vector as inputs. The query vector is constructed using the `models.NamedSparseVector`, which includes the indices and values derived from the query text. This is a crucial step in efficiently retrieving relevant documents. 
+
+Those familiar with the Qdrant API will notice that the extra care taken to be consistent with the existing named vectors API -- this is to make it easier to use sparse vectors in existing codebases. As always, you're still able to apply payload filters, shard keys, and other advanced features you've come to expect from Qdrant.
```

Comment:
this is a repetition from a few lines above?

Reply:
Yes, thanks for catching it!